Library/include/CSP/Multiplayer/Components/CinematicCameraSpaceComponent.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// @file CinematicCameraSpaceComponent.h
+/// @brief Definitions and support for CinematicCamera.
+
+#pragma once
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Multiplayer/ComponentBase.h"
+#include "CSP/Multiplayer/Components/Interfaces/IThirdPartyComponentRef.h"
+#include "CSP/Multiplayer/Components/Interfaces/ITransformComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IEnableableComponent.h"
+
+
+namespace csp::multiplayer {
+
+/// @brief Enumerates the list of properties that can be replicated for a CinematicCamera component.
+enum class CinematicCameraPropertyKeys
+{
+	Position = 0,
+	Rotation,
+	IsEnabled,
+	FocalLength,
+	AspectRatio,
+	SensorSize,
+	NearClip,
+	FarClip,
+	Iso,
+	ShutterSpeed,
+	Aperture,
+	IsViewerCamera,
+	ThirdPartyComponentRef,
+	Num
+};
+
+
+/// @ingroup CinematicCameraSpaceComponent
+/// @brief Data representation of an CinematicCameraSpaceComponent.
+class CSP_API CinematicCameraSpaceComponent : public ComponentBase, public IThirdPartyComponentRef, public ITransformComponent, public IEnableableComponent
```

Comment:
We are currently using the scale component when there is no need. Instead of inheriting the ITransformComponent, we can use IPositionComponent and IRotationComponent.

Reply:
This is what I wasn't sure about.  Without the transform component we don't have the Get/Set Transform functions.  These sound important but are they not used then?

Reply:
Ah, I understand.  If having to Get/Set Transform functions will make it easier for clients, then I'm happy to keep it in, otherwise, it's redundant.

I'm happy for you to decide whether this would be used by clients over Get/Set Position and Rotation

Reply:
It should be fine

Reply:
Its just a helper function to set pos/rot, and scale at the same time. So, if we do not have a scale, its arguably not needed

Reply:
I've removed it